src/frame/mod.rs

```diff
@@ -33,18 +35,31 @@ pub mod traits;
 
 use crate::error;
 
+static STREAM_ID: AtomicI16 = AtomicI16::new(0);
+
+pub type StreamId = i16;
```

Comment:
I'm wondering why we are changing `u16` to `i16`

Reply:
Stream id in cassandra is signed, with negative values reserved.

Reply:
It must be some inconsistency in the spec:

https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v4.spec#L143 says that it can have negative values which are reserved for a server-inited streams.

https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v4.spec#L202 says that `[short]` is an unsigned integer. https://doc.rust-lang.org/std/primitive.u16.html

Anyway, let's maybe stay with your solution that eliminates negative values. 👍